homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -0,0 +1,144 @@
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_ADDRESS,
+    DEFAULT_INTEGRATION_TITLE,
+    DOMAIN,
+    MAX_ADDRESS,
+    MIN_ADDRESS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_and_connect(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    comport = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising com port=%s", comport)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ret[ATTR_MODEL] = f"{client.version()} ({client.pn()})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device=%s", comport)
+        raise err
+    finally:
+        if client.serline.isOpen():
+            client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
+
+
+class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aurora ABB PowerOne."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
+    _defaultcomport = None
+
+    def __init__(self):
+        """Initialise the config flow."""
+        self.config = None
+
+    async def async_step_import(self, conf: dict):
+        """Import a configuration from config.yaml."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
+
+        conf[ATTR_SERIAL_NUMBER] = "sn_unknown_yaml"
+        conf[ATTR_MODEL] = "model_unknown_yaml"
+        conf[ATTR_FIRMWARE] = "fw_unknown_yaml"
+        conf[CONF_PORT] = conf["device"]
+
+        await self.async_set_unique_id(self.flow_id)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(title=DEFAULT_INTEGRATION_TITLE, data=conf)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialised by the user."""
+
+        errors = {}
+        if self._comportslist is None:
+            comports = serial.tools.list_ports.comports(include_links=True)
+            comportslist = []
+            for port in comports:
+                comportslist.append(port.device)
+                _LOGGER.debug("COM port option: %s", port.device)
+            if len(comportslist) > 0:
+                defaultcomport = comportslist[0]
+            else:
+                _LOGGER.warning(
+                    "No com ports found.  Need a valid RS485 device to communicate"
+                )
+                return self.async_abort(reason="no_serial_ports")
+            self._comportslist = comportslist
+            self._defaultcomport = defaultcomport
```

Comment:
This looks like it should be its own function.

You could replace the `self.async_abort` with `data_entry_flow.AbortFlow("no_serial_ports")`

Reply:
Ok, done.  I have kept the return self.async_abort within the calling function because there's a future configuration option that uses TCP/IP and doesn't need to abort if no serial ports are found.